### PR TITLE
Docs: cholfact! double precision only for sparse

### LINKED ISSUE #14810: cholfact for sparse matrix does not keep floating type
In Julia 0.4.3, and Julia 0.5.0-dev, calling cholfact on a sparse Float32 matrix promotes the Cholesky factorization to Float 64. On a dense matrix, it correctly gives a Float32 factorization.

Compare:

```
julia> cholfact(eye(Float32,3))
Base.LinAlg.Cholesky{Float32,Array{Float32,2}} with factor:
3x3 UpperTriangular{Float32,Array{Float32,2}}:
 1.0  0.0  0.0
 0.0  1.0  0.0
 0.0  0.0  1.0
```
With:

```
julia> cholfact(speye(Float32,3))
Base.SparseMatrix.CHOLMOD.Factor{Float64}
type:          LLt
method: simplicial
maxnnz:          3
nnz:             3
```

### PR DESCRIPTION
**Pull request by jiahao**
Closes #14810

Also changes description of `LDL^T` to `LDL'`. The former is wrong for
complex matrices.

### CODE REVIEW

File: base/sparse/cholmod.jl [hunk 1]
```diff
@@ -1248,7 +1248,14 @@ end
         Hermitian{Complex{<:Real},SparseMatrixCSC{Complex{<:Real},SuiteSparse_long}}};
         shift = 0.0) -> CHOLMOD.Factor
 
-Compute the LDLt factorization of `A`, reusing the symbolic factorization `F`.
+Compute the ``LDL'`` factorization of `A`, reusing the symbolic factorization `F`.
```
Comment (andreasnoack) [inline base/sparse/cholmod.jl]: This is `cholfact!`, not `ldltfact`. Something is wrong here. Maybe I've made a copy-paste error when moving the documentation over here.

File: base/sparse/cholmod.jl [hunk 2]
```diff
@@ -1248,7 +1248,14 @@ end
...
+
+** Note **
+
+The computation on a sparse matrix uses the the CHOLMOD library from
+SuiteSparse. CHOLMOD only supports doubles or complex doubles, so input
+matrices not of element type will be converted to `SparseMatrixCSC{Float64}` or
```
Comment (tkelman) [inline base/sparse/cholmod.jl]: not of element type ... ?

File: base/sparse/cholmod.jl (hunk 1 above)
Comment (jiahao) [inline base/sparse/cholmod.jl]: I did vaguely recall you split out ldltfact. So this is just the ordinary Cholesky then?

Comment (andreasnoack) [inline base/sparse/cholmod.jl]: Yes. Here it should just be `Cholesky`, but line 1331 should be fixed like you are doing it here.

**Pull request merged**